### PR TITLE
Minikube guide updates

### PR DESCRIPTION
**Pull request by aditighag**
Add a note about `--cni=cilium` flag that can automatically install Cilium. However, it installs v1.8 by default so add a note regarding this.

Cilium init script mounts the BPF fs if not already mounted so the manual step to mount BPF fs isn't required. Let's double check - 

```
$ minikube start --network-plugin=cni
$ cilium install
$ minikube ssh -- mount | grep bpf
none on /sys/fs/bpf type bpf (rw,relatime)

ks logs cilium-zp2st -c ebpf-mount
Mounting eBPF filesystem...
```

`ebpf-mount` is the init container that mounts the BPF fs. Here is the relevant code in cilium/cli - https://github.com/cilium/cilium-cli/blob/master/install/install.go#L685

Fixes: #16324

### CODE REVIEW

File: Documentation/gettingstarted/k8s-install-default.rst [hunk 1]
```diff
@@ -98,7 +98,13 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
        .. code-block:: shell-session
 
           minikube start --network-plugin=cni
-          minikube ssh -- sudo mount bpffs -t bpf /sys/fs/bpf
+
+.. note::
+
+   From minikube v1.12.1+, cilium networking plugin can be enabled directly with
```
Comment (aanm) [inline Documentation/gettingstarted/k8s-install-default.rst]: Let's bump the minikube version in line 95 to v1.12.1 and remove this note altogether.

**Review by aanm — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 1 above)
Comment (aditighag) [reply Documentation/gettingstarted/k8s-install-default.rst]: Yeah, thought about it. Sounds good to me!

File: Documentation/gettingstarted/k8s-install-default.rst [hunk 2]
```diff
@@ -92,14 +92,22 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
...
           minikube ssh -- sudo mount bpffs -t bpf /sys/fs/bpf
 
+.. note::
+
+   From minikube v1.12.1+, cilium networking plugin can be enabled directly with
+   ``--cni=cilium`` parameter in ``minikube start`` command. With this
+   flag enabled, ``minikube`` will not only mount eBPF file system but also
+   deploy ``quick-install.yaml`` automatically. However, this may not install the
+   latest version of cilium.
+
```
Comment (aanm) [inline Documentation/gettingstarted/k8s-install-default.rst]: I don't think we should have it, it's misleading and users might think it's an easier path to install Cilium, which it is, but installs an older Cilium version (1.8)

**Review by aanm — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/k8s-install-default.rst [hunk 3]
```diff
@@ -92,14 +92,22 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
...
 
-       Install minikube >= v1.5.2 as per minikube documentation: 
+       Install minikube >= v1.5.2 as per minikube documentation:
        `Install Minikube <https://kubernetes.io/docs/tasks/tools/install-minikube/>`_.
 
        .. code-block:: shell-session
 
-          minikube start --network-plugin=cni
+          minikube start --network-plugin=cni --memory=4096
           minikube ssh -- sudo mount bpffs -t bpf /sys/fs/bpf
```
Comment (aditighag) [inline Documentation/gettingstarted/k8s-install-default.rst]: Let's double check if this step is required. Cilium init script mounts the BPF fs if not already mounted. 
I just ran these commands - 

```
$ minikube start --network-plugin=cni
$ cilium install
$ minikube ssh -- mount | grep bpf
none on /sys/fs/bpf type bpf (rw,relatime)


```

Comment (nbusseneau) [reply Documentation/gettingstarted/k8s-install-default.rst]: It should not, this is done via `cilium install` now: https://github.com/cilium/cilium-cli/blob/ce6aa0252795e5ec4909f39245bced0cc4cbdf6f/install/install.go#L685

Comment (aditighag) [reply Documentation/gettingstarted/k8s-install-default.rst]: That's already mentioned in the commit description. ;)

Comment (aanm) [reply Documentation/gettingstarted/k8s-install-default.rst]: I've tested it locally. It works now, I'm not sure why it failed for me before :thinking:

**Review by qmonnet — APPROVED** [review]
> Cilium init script mounts the BPF fs if not already mounted so the manual step to mount BPF fs isn't required.

(Note that it could be worth having the above in the commit log of the revert patch. Not blocking, though.)

Looks good, thank you! Don't forget to sign-off :)

File: Documentation/gettingstarted/k8s-install-default.rst [hunk 4]
```diff
@@ -98,7 +98,12 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
        .. code-block:: shell-session
 
           minikube start --network-plugin=cni
-          minikube ssh -- sudo mount bpffs -t bpf /sys/fs/bpf
+
+.. note::
+
+   From minikube v1.12.1+, cilium networking plugin can be enabled directly with
+   ``--cni=cilium`` parameter in ``minikube start`` command. However, this may not
+   install the latest version of cilium.
```
Comment (aanm) [inline Documentation/gettingstarted/k8s-install-default.rst]: I don't think we should have it, it's misleading and users might think it's an easier path to install Cilium, which it is, but installs an older Cilium version (1.8)

**Review by aanm — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 2 above)
Comment (aditighag) [reply Documentation/gettingstarted/k8s-install-default.rst]: I've already added a note about the version getting installed. It's useful to have a simpler step for users who don't care about version, and/or might miss the manual step.
Also, check this note from minikube command - 

>❗  With --network-plugin=cni, you will need to provide your own CNI. See --cni flag as a user-friendly alternative

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 4 above)
Comment (aditighag) [reply Documentation/gettingstarted/k8s-install-default.rst]: I've already added a note about the version getting installed. It's useful to have a simpler step for users who don't care about version, and/or might miss the manual step.
Also, check this note from minikube command -

❗ With --network-plugin=cni, you will need to provide your own CNI. See --cni flag as a user-friendly alternative

**Comment by aditighag** [comment]
@qmonnet Updated the commit description for the reverted commit. Thanks!

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 4 above)
Comment (joestringer) [reply Documentation/gettingstarted/k8s-install-default.rst]: Given this is just a note to basically acknowledge that there is another path (but suggest why we don't advise that path by default), I'm OK with this. However I think it should be hidden inside the minikube group-tab, not at the outermost level of indentation.

**Review by joestringer — CHANGES_REQUESTED** [review]
👍 

I'm ambivalent about documenting the `minikube --cni=cilium ...` flag or not, seems like we have legitimate reasons not to recommend the user to use it so it'd be simpler to omit the instruction. That said it's not a big deal to also just let the user know like this. However I think if we do so, we should move the note inside the group tab for minikube users rather than instruct all users about this, even if they've clicked one of the other group tabs (why would cloud or kind users need to be told about a minikube feature?).

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 4 above)
Comment (aditighag) [reply Documentation/gettingstarted/k8s-install-default.rst]: @joestringer Good catch! Fixed the formatting, PTAL. Thanks!

**Pull request merged**